groth16/src/prover.rs

```diff
@@ -272,3 +266,25 @@ where
         c: g_c.into_affine(),
     })
 }
+
+fn calculate_coeff<G: AffineCurve>(
```

Comment:
Removed duplicate code and abstracted to function - reduces error chances from copy-pasting

Reply:
Is there maybe a better name for this function? Trying to think where also we might see it, beyond groth16 or gm17